### ISSUE TITLE
Is there a way to reset form to default values if they're changed?

### ISSUE THREAD

**Issue by netknight** [edmundhung/conform#129]
For example I'm setting form like this:
```
const [form, fields] = useForm<PostUpdate>({
    defaultValue: defaultValue,
    lastSubmission: lastSubmission,
    initialReport: "onBlur"
  });
```

Both `defaultValue` & `lastSubmission` are component props that are passed from another component.
When `defaultValue` changes - nothing changes in form, I've tried setting input 2 ways:
1. `<input type="text" {...conform.input(field)} /> `
2. `<input type="text" name={field.name} defaultValue={field.defaultValue} /> `

It looks like form is not being reset if defaultValue changes. Maybe there are any ways to reset it manually to default value?

I've also tried to add useEffect hook:
```
  const defaultValueDeferred = React.useDeferredValue(defaultValue);
  React.useEffect(() => {
    form.ref.current?.reset();
  }, [defaultValueDeferred, form.ref]);
```

But it resets to a previous state of the form instead of new one.

And I'm using Remix, so defaultValues are retrieved val loader & lastSubmission via action (maybe problem relates to Remix integration).

**Comment by edmundhung** [comment]
Hi @netknight,

You are very close to the solution. Conform will not update its internal reference of the `defaultValue` until a reset event is triggered. So calling `.reset()` with useEffect should update the `defaultValue` it is passing on each field.

> But it resets to a previous state of the form instead of new one.

This is actually a react problem. The reset event will only brings the inputs back to its initial value which is set only the first time it is rendered by react. So passing a new `defaultValue` has no effect. A good solution is to make React remount the elements by specifying a `key`.

```tsx
<input type="text" key={field.defaultValue} {...conform.input(field)} /> 
```

**Comment by netknight** [comment]
Tried playing with key but this does nothing, because I clearly see that these variables aren't updated after `defaultValue` changes:
```
const [form, fields] = useForm(...)
```
I just added `console.log` after `useEffect` hook:
```
React.useEffect(() => {
    form.ref.current?.reset();
  }, [defaultValue, form.ref]);

  console.log(fields.slug.defaultValue, defaultValue?.slug);
```
and I see different values in console!

If you know how to achieve this. Can you provide a working example? Anyway it would be nice to have such example in documentation, because it is very common use-case.

**Comment by edmundhung** [comment]
I can reproduce the issue you had [here](https://codesandbox.io/s/conform-default-value-9v0ei5?file=/app/routes/colors.%24color.tsx). The form value is not reset unless you click on the same link twice.

This is probably a bug on the reset event handler. I will put up a bug fix later today. I agree this is a fairly common use-case. Thanks for reporting it 😃

**Comment by netknight** [comment]
> The form value is not reset unless you click on the same link twice.

Moreover, I've noticed some strange behaviour: Even when you click same link twice - some fields may not update at all. I wasn't able to understand what this depend on.

> I will put up a bug fix later today.

I would recommend to think about including automatic reset into the library. I believe most of user will expect exactly this behaviour. I had an experience with react-hook-form and it doesn't have such a problem.

**Comment by edmundhung** [comment]
This should be resolved with [v0.6.1](https://github.com/edmundhung/conform/releases/tag/v0.6.1). 

You can check the updated example [here](https://codesandbox.io/s/conform-default-value-9v0ei5?file=/app/routes/colors.%24color.tsx). Tere is no need to call reset now. But setting the `key` to remount each input is still required.

**Comment by ivan-kleshnin** [comment]
Any tip for people not using React actions? `lastResult` type is heavily action specific.

Is it possible to extend it to also accept `DefaultValue<Schema>` so form would adopt new default value automatically?
